### PR TITLE
Shooter branch

### PR DESCRIPTION
**Pull request by TapChap**

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -55,8 +55,12 @@ public enum Modules {
                 this.DRIVE_MOTOR_REVERSED = DRIVE_MOTOR_REVERSED;
                 this.SPIN_MOTOR_REVERSED = SPIN_MOTOR_REVERSED;
             }
+
```
Comment (TapChap) [inline src/main/java/frc/robot/Constants.java:58]:
```suggestion
```

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -55,8 +55,12 @@ public enum Modules {
...
         }
 
+
+
+
```
Comment (TapChap) [inline src/main/java/frc/robot/Constants.java:63]:
```suggestion
```

File: src/main/java/frc/robot/Constants.java [hunk 3]
```diff
@@ -138,4 +142,14 @@ public static class LedsConstants {
...
     }
+    public static final class ShooterConstants {
+        public static final int LEADER_SHOOTER_MOTOR_ID = 30;
+        public static final int FOLLOWER_SHOOTER_MOTOR_ID = 31;
+        public static final int LEADER_LINEAR_SHOOTER_MOTOR_ID = 31;
+        public static final int FOLLOWER_LINEAR_SHOOTER_MOTOR_ID = 31;
+
+        public static final int SHOOTER_BEAMBREAK_CHANNEL = 32;
+        public static final double SHOOT_SPEED = 0;
+
```
Comment (TapChap) [inline src/main/java/frc/robot/Constants.java:153]:
```suggestion
```

File: src/main/java/frc/robot/Constants.java [hunk 4]
```diff
@@ -138,4 +142,14 @@ public static class LedsConstants {
         public static final int LENGTH = 0;
 
     }
+    public static final class ShooterConstants {
+        public static final int LEADER_SHOOTER_MOTOR_ID = 30;
+        public static final int FOLLOWER_SHOOTER_MOTOR_ID = 31;
+        public static final int LEADER_LINEAR_SHOOTER_MOTOR_ID = 31;
```
Comment (TapChap) [inline src/main/java/frc/robot/Constants.java:148]: "LEADER_LINEAR_SHOOTER_MOTOR_ID" ??

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    public Shooter(){
+        followerShooterMotor.follow(shooterMotors, true);
+        linearFollowerMotor.follow(linearLeader);
+    }
+
+    private final void setLinear(double speed){
+        linearLeader.set(speed);
+    }
+
+    public Command LinearStart(DoubleSupplier speed){
```
Comment (TapChap) [inline src/main/java/frc/robot/subsystems/Shooter.java:39]: "LinearStart" ?

Comment (Korngut) [reply src/main/java/frc/robot/subsystems/Shooter.java:39]: Fixed

File: src/main/java/frc/robot/Constants.java (hunk 4 above)
Comment (Korngut) [reply src/main/java/frc/robot/Constants.java:148]: Fixed

**Pull request closed without merging**